network/ethernet_frame.ksy

```diff
@@ -48,6 +44,20 @@ instances:
       first location bears special marker (0x8100), then it is not the
       real ether frame yet, an additional payload (`tci`) is expected
       and real ether type is upcoming next.
+  body:
+    io: _root._io
+    pos: _root._io.size - body_and_checksum.length
+    size: body_and_checksum.length - 4
```

Comment:
I think this is an unnecessarily clumsy and complex implementation. Why not just

```diff
   - id: body
-    size-eos: true
+    size: (_io.size - _io.pos) - checksum._sizeof
     type:
       switch-on: ether_type
       cases:
         'ether_type_enum::ipv4': ipv4_packet
         'ether_type_enum::ipv6': ipv6_packet
+  - id: checksum
+    size: 4
```

...?

Reply:
Oh, this is indeed a much better solution.